arch/xtensa/core/coredump.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <debug/coredump.h>
+#include <xtensa-asm2.h>
+
+#define ARCH_HDR_VER			1
+#define XTENSA_BLOCK_HDR_VER	1
+
+enum xtensa_soc_code {
+	XTENSA_SOC_UNKNOWN = 0,
+	XTENSA_SOC_SAMPLE_CONTROLLER,
+	XTENSA_SOC_ESP32,
+};
+
+struct xtensa_arch_block {
+	/* Each Xtensa SOC can omit registers (e.g. loop
+	 * registers) or assign different index numbers
+	 * in xtensa-config.c. GDB identifies registers
+	 * based on these indices
+	 *
+	 * (This must be the first field or the GDB server
+	 * won't be able to unpack the struct while parsing)
+	 */
+	uint8_t		soc;
+
+	/* Future versions of Xtensa coredump
+	 * may expand minimum set of registers
+	 *
+	 * (This should stay the second field for the same
+	 * reason as the first once we have more versions)
+	 */
+	uint16_t	version;
+
+	struct {
+		/* Minimum set shown by GDB 'info registers',
+		 * skipping user-defined register EXPSTATE
+		 *
```

Comment:
For completeness, you might want to include the various MISCn (one per XCHAL_NUM_MISC_REGS) and EXCSAVEn (one for each XCHAL_NUM_INTERRUPTS).  The Zephyr kernel code uses these for various purposes, the big one being the struct cpu pointer for the current CPU.  That may take some design thought, though, because the actual register used is different for different platforms...  Future work, I guess.